lib/cursor/aggregationCursor.js

```diff
@@ -57,21 +58,25 @@ util.inherits(AggregationCursor, Readable);
 function _init(model, c, agg) {
   if (!model.collection.buffer) {
     model.hooks.execPre('aggregate', agg, function() {
+      Object.assign(c.options, agg._optionsForExec());
```

Comment:
Similar to previous comments, is there any need to set `c.options` at all? Could just call `aggregate(agg._pipeline, agg.optionsForExec())`

Reply:
In QueryCursor, options is also set to a new object, so I followed the same approach here.
That said, QueryCursor uses the options object for some checks while AggregateCursor only uses it in _init, so it might make sense to remove it from AggregateCursor for now.